sources/filesystem/__init__.py

```diff
@@ -76,11 +93,11 @@ def filesystem(
     if isinstance(credentials, AbstractFileSystem):
         fs_client = credentials
     else:
-        fs_client = fsspec_filesystem(bucket_url, credentials)[0]
+        fs_client = fsspec_filesystem(bucket_url, credentials, kwargs=kwargs)[0]
 
     files_chunk: List[FileItem] = []
     for file_model in glob_files(fs_client, bucket_url, file_glob):
-        file_dict = FileItemDict(file_model, credentials)
+        file_dict = FileItemDict(file_model, fs_client)
```

Comment:
ok now I see: we should not pass `credentials` to `FileItemDict`, we should pass
`FilesystemConfiguration(protocol, credentials, kwargs=kwargs, client_kwargs=client_kwargs)`

so that would require changing your branch to the core and in the `FileDict` implementation change:
```
@property
    def fsspec(self) -> AbstractFileSystem:
```
to use: `fsspec_from_config`

looks like big improvement: we can handle parametrized filesystems now

Reply:
Looks like a great idea.  I'll try it

Reply:
> it often happens that file items are processed in different threads and I'd avoid sharing a single instance of filesystem.

@rudolfix   Note that fsspec  instances are cacheable by default.    (It is separate to the dir listings cache).    It can be disabled by passing `skip_instance_cache=True` on instantiation.   Shall I hard-code disabling it in this PR?    Or should it be separate issue?

Reply:
I have got this working locally.   

Question about the imports in verified-sources `sources/filesystem/__init__.py`

```
from dlt.sources.filesystem import FileItem, FileItemDict, fsspec_filesystem, glob_files
from dlt.sources.credentials import FileSystemCredentials
from dlt.common.storages.fsspec_filesystem import fsspec_from_config
from dlt.common.storages.configuration import FilesystemConfiguration
```

I added the last two lines.   But should I be importing them via `dlt.sources.<some_module>` instead?     What is the reason for `dlt.sources.<some_module>` ?

Also I no longer need to import `fsspec_filesystem` (line 2) .  Do I remove it in dlt.sources.filesystem too?